minio/minio
# fix: fetchLambdaInfo should return consistent results

## Description

fetchLambdaInfo should return a complete list of configured targets irrespective to the errors while initializing the targets.

- Introduced a function `FetchRegisteredTargets` which will return
  a complete set of registered targets irrespective to their states,
  if the `returnOnTargetError` flag is set to `False`
- Refactor NewTarget functions to return non-nil targets

## Motivation and Context

- `mc admin info --json` should not return empty notification list ('notifications': []) if any of configured target is offline
- In this case, fallback on target errors while initializing the targets for preparing a targetList (always add a target to the TargetList)

## How to test this PR?
- Configure multiple targets
- toggle the targets from offline -><- online
- `mc admin info --json` should always give a non-empty notification list and should say if the configured targets are online/offline

For eg, 

- Run a webhook on :8080
- Run a redis server
- Add webhook and redis endpoints in MinIO
- Make redis server offline
- Check `mc admin info --json`, It should provide a notification list with two targets and also stating redis target as "offline"

```
"notifications": [
                {
                    "redis": [
                        {
                            "1": {
                                "status": "Offline"
                            }
                        }
                    ]
                },
                {
                    "webhook": [
                        {
                            "1": {
                                "status": "Online"
                            }
                        }
                    ]
                }
            ]
```

## Types of changes

## Checklist:
